app/controllers/admin/pages_controller.rb

```diff
@@ -39,5 +46,51 @@ def page_params
                           social_image]
       params.require(:page).permit(allowed_params)
     end
+
+    def prepopulate_new_form(slug)
```

Comment:
We pre-populate the new Page record with some information to enhance the user experience for the admin

Reply:
The reason this function seems a bit hardcoded is because I can't pass the html as a param as it's a bit dodge :). Hence, passing the slug and then setting the necessary params seems like the best way to do this. this PR is a high priority and needs to be merged by Monday (if there are no fundamental issues), however if you do have suggestions on refactoring this is some more dynamic way I'd love to chat more and perhaps we can send out a second iteration.

Reply:
Thanks for pointing this out! Definitely would like to see this a bit more extracted/generalized, but it's perfectly fine for now given the constraints you listed!